### PR TITLE
[JSON]Fix filtering in GetSongs and GetAlbums

### PR DESCRIPTION
**Pull request by DaveTBlake**
Fix path filtering in **AudioLibrary::GetSongs** and filtering by playcount, lastplayed or dateadded in **AudioLibrary::GetAlbums** after changes of #14160.

JSON API version bump patch as only internal processing change.

Can be tested with JSON commands:
```
{"jsonrpc": "2.0", "id": 1, "method": "AudioLibrary.GetAlbums", 
"params": { "filter": {"operator": "is", "field": "playcount", "value": "1"},
"properties": [ "lastplayed"] }}
```

```
{"jsonrpc":"2.0","id":1,"method":"audioLibrary.GetSongs",
"params":{"limits": {"start": 0, "end": 600},
"filter": {"operator": "contains", "field": "path", "value": "bruckner"},
"properties":["displayartist", "file"]}}
```

### CODE REVIEW

File: xbmc/music/MusicDatabase.cpp [hunk 1]
```diff
@@ -5958,19 +5973,33 @@ bool CMusicDatabase::GetSongsByWhereJSON(const std::set<std::string>& fields, co
...
       return false;
 
+    std::string strSQLExtra;
+    if (!BuildSQL(strSQLExtra, extFilter, strSQLExtra))
+      return false;
+
+    // Count number of songs that satisfy selection criteria 
+    // (includes xsp limits from filter, but not sort limits)
+    // Use songview as filter rules in where clause may use album and path JOIN fields
+    total = (int)strtol(GetSingleValue("SELECT COUNT(1) FROM songview " + strSQLExtra, m_pDS).c_str(), NULL, 10);
```
Comment (Rechi) [inline xbmc/music/MusicDatabase.cpp]: C++ style cast

**Comment by zag2me** [comment]
Tested on windows and confirmed this fixes the SQL error I found while using the Artwork beef Add-on.

Thanks!

**Comment by DaveTBlake** [comment]
Thanks for testing @MilhouseVH, darn substandard MySQL strikes again!
Anyway I have adjusted GetAlbums to handle it, only "albums" has scalar subquery fields as possible filter criteria. Could you test GetAlbums again please, in case I have slipped again.

**Comment by MilhouseVH** [comment]
I've tested the two queries in this PR again, and the MySQL issue is resolved. I've also repeated the testing from the previous PR, and those queries continue to return the expected results. Many thanks @DaveTBlake, this looks good to me.

**Pull request merged**